mdagent/tools/base_tools/analysis_tools/sasa.py

```diff
@@ -0,0 +1,154 @@
+
+        Parameters:
+        path_registry (PathRegistry): mapping file IDs to file paths.
+        top_fileid (str): File ID for the topology file.
+        traj_fileid (str, optional): File ID for the trajectory file.
+        mol_name (str, optional): Name of the molecule or protein.
+        """
+        self.path_registry = path_registry
+        all_fileids = self.path_registry.list_path_names()
+        if top_fileid not in all_fileids:
+            raise ValueError("Topology File ID not found in path registry")
+        top_path = self.path_registry.get_mapped_path(top_fileid)
+
+        if traj_fileid:
+            if traj_fileid not in all_fileids:
+                raise ValueError("Trajectory File ID not found in path registry")
+            traj_path = self.path_registry.get_mapped_path(traj_fileid)
+            self.traj = md.load(traj_path, top=top_path)
+        else:
+            self.traj = md.load(top_path)
+        self.molecule_name = mol_name if mol_name else top_fileid.replace("top_", "")
+        self.sasa = None
+        self.residue_sasa = None
+
+    def calculate_sasa(self, probe_radius=0.14):
+        """
+        Calculate the Solvent Accessible Surface Area (SASA) for each
+        frame in the trajectory using Shrake-Rupley algorithm. Hydrogens
+        are excluded for SASA residues calculation.
+
+        Parameters:
+        probe_radius (float, optional): The radius of the probe used to calculate SASA.
+            Default is 0.14 nm (1.4 Å).
+        """
+        self.sasa = md.shrake_rupley(self.traj, probe_radius=probe_radius, mode="atom")
+        self.residue_sasa = md.shrake_rupley(
+            self.traj, probe_radius=probe_radius, mode="residue"
+        )
+
+        # save to file
+        sasa_file = f"{self.path_registry.ckpt_records}/sasa_{self.molecule_name}.csv"
+        i = 0
+        while os.path.exists(sasa_file):
+            i += 1
+            sasa_file = (
+                f"{self.path_registry.ckpt_records}/sasa_{self.molecule_name}_{i}.csv"
+            )
+        np.savetxt(sasa_file, self.sasa, delimiter=",", header="SASA (nm²)")
+        self.path_registry.map_path(
+            f"sasa_{self.molecule_name}_{i}",
+            sasa_file,
+            description=f"SASA values for the molecule {self.molecule_name}",
+        )
+        return f"SASA values computed and saved to {sasa_file}"
+
+    def plot_sasa(self):
+        """
+        Plot the total SASA and per-residue SASA over time.
+
+        Returns:
+        None
+        """
+        message = ""
+        if self.sasa is None or self.residue_sasa is None:
+            message += self.calculate_sasa()
+
+        # Returns areasnp.array, shape=(n_frames, n_features)
+        # if there's only one frame, don't plot
+        if self.sasa.ndim == 1:
+            message += " Only one frame in trajectory. No SASA plot generated."
+            return message
```

Comment:
we can also add that only value right?

```suggestion
            message += f"Available Surface area is: {self.residue_sasa}. 
            return message
```

Reply:
actually, if you have like 300 residues, it'll return 300 values. I instead computed & returned "total_sasa", which is scalar.